### PR TITLE
[CHORE] Update eslint to v6

### PR DESCRIPTION
**Pull request by Gaurav0**
1. Update ESLint to v6
2. ~~Resolve to latest version of babel-scope to avoid flagging false positives~~
3. ~~Install eslint-plugin-autofix and autofix no-undef-var errors~~
4. ~~Allow tests to have no-undef lint rule not pass~~
5. Disable some new rules that would take time to fix
6. Fix remaining ESLint errors
7. Run sync-dev-dependency

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "build": "yarn workspace ember-data build",
     "build:production": "yarn workspace ember-data build:production",
     "lint:js": "./bin/lint-js",
-    "lint:prettier": "eslint --plugin prettier .",
+    "lint:prettier": "node --max-old-space-size=4096 ./node_modules/.bin/eslint --plugin prettier .",
```
[the remark this answers is not in the document]
Comment (Gaurav0) [reply package.json]: @rwjblue @runspired This seems to have been fixed somehow. Removing.

File: .eslintrc.js [hunk 2]
```diff
@@ -31,6 +37,10 @@ module.exports = {
     'no-irregular-whitespace': 'error',
     'no-undef': 'error',
     'no-eq-null': 'error',
+
+    // probably want to fix these new rules later in separate PR
```
[the remark this answers is not in the document]
Comment (Gaurav0) [reply .eslintrc.js]: Done. See #6405

File: packages/adapter/package.json [hunk 3]
```diff
@@ -55,4 +55,4 @@
   "ember-addon": {
     "configPath": "tests/dummy/config"
   }
-}
+}
```
Comment (HeroicEric) [inline packages/adapter/package.json]: Why were these newlines removed?

Comment (Gaurav0) [reply packages/adapter/package.json]: Fixed.

File: package.json [hunk 4]
```diff
@@ -110,14 +111,23 @@
     "rsvp": "^4.8.5",
     "semver": "^6.2.0",
     "silent-error": "^1.1.1",
-    "typescript": "~3.5.3"
+    "typescript": "~3.5.3",
+    "@ember-data/-test-infra": "3.14.0-alpha.1",
+    "babel-plugin-debug-macros": "^0.3.2",
```
[the remark this answers is not in the document]
Comment (rwjblue) [reply package.json]: This is still somewhat of an open issue, but I don't think it needs to block this PR specifically.

**Pull request merged**